### PR TITLE
CDPS-151: Add role based permissions to Update Alert link

### PR DESCRIPTION
**Pull request by davidmiddletonmoj**

### CODE REVIEW

File: server/controllers/alertsController.ts [hunk 1]
```diff
@@ -34,8 +36,14 @@ export default class AlertsController {
     // Get prisoner data for banner and for use in alerts generation
     const prisonerData = await this.prisonerSearchService.getPrisonerDetails(req.params.prisonerNumber)
 
+    // Set role based permissions
+    const { authorities: roles = [] } = jwtDecode(res.locals.user.token) as { authorities?: string[] }
+    const canUpdateAlert =
+      roles.some(role => role === Role.UpdateAlert) &&
+      (res.locals.user.activeCaseLoadId === prisonerData.prisonId || ['OUT', 'TRN'].includes(prisonerData.prisonId))
```
Comment (danielburnley) [inline server/controllers/alertsController.ts:43]: Hmmm - The API when dealing with these two prison (/agency) IDs calls out a particular role (`INACTIVE_BOOKINGS`) - https://github.com/ministryofjustice/prison-api/blob/main/src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java#L541-L543

Is that something we might want to care about here? I'm not sure

Comment (danielburnley) [inline server/controllers/alertsController.ts:43]: I wonder if these three lines could be made into a function that takes a role as an argument and returns true/false

Comment (davidmiddletonmoj) [reply server/controllers/alertsController.ts:43]: I thought about that, but the next tickets on Case Notes use slightly different logic, so there'd be no reuse (at least for now). If a few more tickets use the exact logic, then would be worth doing.

Comment (davidmiddletonmoj) [reply server/controllers/alertsController.ts:43]: Good question - I don't know in this case. The Case Notes roles logic does look for that role and those IDs, so guessing it's fine here as-is, but we'll see if anything comes out of testing with more realistic cases/roles.

Comment (danielburnley) [reply server/controllers/alertsController.ts:43]: Ah cool - yeah lets see how the rest shake out at least

**Review by danielburnley — APPROVED** [review]

**Pull request merged**